examples/middleware_factory/src/combining_powertools_utilities_function.py

```diff
@@ -0,0 +1,121 @@
+import json
+from typing import Callable
+
+import boto3
+import combining_powertools_utilities_schema as schemas
+import requests
+
+from aws_lambda_powertools import Logger, Tracer
+from aws_lambda_powertools.event_handler import APIGatewayRestResolver
+from aws_lambda_powertools.event_handler.exceptions import InternalServerError
+from aws_lambda_powertools.middleware_factory import lambda_handler_decorator
+from aws_lambda_powertools.shared.types import JSONType
+from aws_lambda_powertools.utilities.feature_flags import AppConfigStore, FeatureFlags
+from aws_lambda_powertools.utilities.jmespath_utils import extract_data_from_envelope
+from aws_lambda_powertools.utilities.typing import LambdaContext
+from aws_lambda_powertools.utilities.validation import SchemaValidationError, validate
+
+app = APIGatewayRestResolver()
+tracer = Tracer()
+logger = Logger()
+
+table_historic = boto3.resource("dynamodb").Table("HistoricTable")
+
+app_config = AppConfigStore(environment="dev", application="comments", name="features")
+feature_flags = FeatureFlags(store=app_config)
+
+
+@lambda_handler_decorator(trace_execution=True)
+def middleware_custom(handler: Callable, event: dict, context: LambdaContext):
+
+    # validating the INPUT with the given schema
+    # X-Customer-Id header must be informed in all requests
+    try:
+        validate(event=event, schema=schemas.INPUT)
+    except SchemaValidationError as e:
+        return {
+            "statusCode": 400,
+            "body": json.dumps(str(e)),
+        }
+
+    # extracting headers and requestContext from event
+    headers = extract_data_from_envelope(data=event, envelope="headers")
+    request_context = extract_data_from_envelope(data=event, envelope="requestContext")
+
+    logger.debug(f"X-Customer-Id => {headers.get('X-Customer-Id')}")
+    tracer.put_annotation(key="CustomerId", value=headers.get("X-Customer-Id"))
+
+    response = handler(event, context)
+
+    # automatically adding security headers to all responses
+    # see: https://securityheaders.com/
+    logger.info("Injecting security headers")
+    response["headers"]["Referrer-Policy"] = "no-referrer"
+    response["headers"]["Strict-Transport-Security"] = "max-age=15552000; includeSubDomains; preload"
+    response["headers"]["X-DNS-Prefetch-Control"] = "off"
+    response["headers"]["X-Content-Type-Options"] = "nosniff"
+    response["headers"]["X-Permitted-Cross-Domain-Policies"] = "none"
+    response["headers"]["X-Download-Options"] = "noopen"
```

Comment:
oh boy this could totally be its own utility class for Powertools

Reply:
Yea! We could add this as a utility and make things even safer!